services/user/XAdmin/ui/src/hooks/useKeyDevices.ts

```diff
@@ -0,0 +1,56 @@
+import { useQuery, useMutation } from "@tanstack/react-query";
+
+import { queryKeys } from "@/lib/queryKeys";
+import { chain } from "@/lib/chainEndpoints";
+import { exportKeyToDER, generateP256Key } from "@/lib/keys";
+
+export const useKeyDevices = () =>
+    useQuery({
+        queryKey: queryKeys.keyDevices,
+        queryFn: async () => {
+            try {
+                return chain.getKeyDevices();
```

Comment:
By returning this promise without `await` I think it's gonna skip over the `catch` logic you have. 
```suggestion
                return await chain.getKeyDevices();
```

Reply:
Good catch. Thanks.